app/models/listing.rb

```diff
@@ -32,6 +33,12 @@ class Listing < ApplicationRecord
   validate :restrict_markdown_input
   validate :validate_tags
 
+  # [@atsmith813] this is adapted from the `listing_search` property in
+  # `config/elasticsearch/mappings/listings.json`
+  pg_search_scope :search_listings,
+                  against: %i[body_markdown cached_tag_list location slug title],
+                  using: { tsearch: { prefix: true } }
```

Comment:
I have good news and bad news on this.

The bad news is that this query is slower than the ElasticSearch version against a prod snapshot by a factor of 2-3x.

The good news is that it still executes in ~90ms because the table only contains a few thousand records (not bad considering the complexity of the query) and traffic to the endpoint that invokes this search is low enough that it shouldn't matter (1 request per ~2 seconds). I'm happy to let this one go live without the search index and re-evaluate later whether we need it.

I'm mainly pointing it out here so that if someone asks later why this doesn't have an index (including my forgetful self), drop a link to this comment. 😄